patterns.md

```diff
@@ -143,4 +152,53 @@ passed as a single object. Hence when there's a collection of different objects,
 we pass a dictionary containing them all.
 
 
+## Application
+
+### Layered approach
+
+An application should be split into three layers, each with their own role:
```

Comment:
Should we say something about dependency flow? I'd describe our flavour of layers as 'open layers', where higher layers can import from any layer lower down. We also are a bit more relaxed about imports between interfaces and domain.

Reply:
Yes, true. Have expanded a little in https://github.com/octoenergy/styleguide/commit/ece9dd2189bac2ba8d61858bd0d05d58c181719b